src/Etsy/Provider.php

```diff
@@ -2,21 +2,74 @@
 
 namespace SocialiteProviders\Etsy;
 
-use SocialiteProviders\Manager\OAuth1\AbstractProvider;
-use SocialiteProviders\Manager\OAuth1\User;
+use SocialiteProviders\Manager\OAuth2\AbstractProvider;
+use SocialiteProviders\Manager\OAuth2\User;
 
 class Provider extends AbstractProvider
 {
-    public const IDENTIFIER = 'ETSY';
+    /**
+     * Unique Provider Identifier.
+     */
+    const IDENTIFIER = 'ETSY';
 
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopes = ['email_r'];
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $scopeSeparator = ' ';
+
+    /**
+     * {@inheritdoc}
+     */
+    protected $usesPKCE = true;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getAuthUrl($state)
+    {
+        return $this->buildAuthUrlFromBase('https://www.etsy.com/oauth/connect', $state);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getTokenUrl()
+    {
+        return 'https://openapi.etsy.com/v3/public/oauth/token';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function getUserByToken($token)
+    {
+        $tokenData = explode('.', $token);
+        $response = $this->getHttpClient()->get('https://openapi.etsy.com/v3/application/users/'.$tokenData[0], [
+            'headers' => [
```

Comment:
Can you use instead `\GuzzleHttp\RequestOptions::HEADERS` constant here instead ?

Reply:
Updated!